go/couler/submitter/argo_submitter.go

```diff
@@ -0,0 +1,55 @@
+package submitter
+
+import (
+	"fmt"
+	metav1 "k8s.io/apimachinery/pkg/apis/meta/v1"
+	"k8s.io/apimachinery/pkg/fields"
+	"k8s.io/client-go/tools/clientcmd"
+
+	wfv1 "github.com/argoproj/argo/pkg/apis/workflow/v1alpha1"
+	wfclientset "github.com/argoproj/argo/pkg/client/clientset/versioned"
+)
+
+// ArgoWorkflowSubmitter holds configurations used for workflow submission
+type ArgoWorkflowSubmitter struct {
+	namespace      string
+	kubeConfigPath string
+}
+
+// Submit takes an Argo Workflow object and submit it to Kubernetes cluster
+func (submitter *ArgoWorkflowSubmitter) Submit(wf wfv1.Workflow) (wfv1.Workflow, error) {
```

Comment:
Here we should have a way to return immediately after submitting the workflow successfully. Image run this in a service.

Reply:
Thanks. Addressed this in #138.